scripts/strict-null-checks-ci.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * A script for formatting a Markdown report for CI on the number of strictNullChecks errors we're
+ * seeing on the current branch vs on main. The report also includes some info about the most
+ * error-filled files, as well as the errors we see most often.
+ */
+const fs = require('fs');
+
+/**
+ * Load JSON data, formatted by `tsc-output-parser`
+ *
+ * In the GH Actions workflow we write two json files, one for the PR
+ * branch and one for main (so that we can compare them).
+ *
+ */
+const prData = JSON.parse(String(fs.readFileSync('./null_errors_pr.json')));
+const mainData = JSON.parse(String(fs.readFileSync('./null_errors_main.json')));
+
+const errorsOnMain = mainData.length;
+const errorsOnPR = prData.length;
+
+/**
+ * Build up an object which maps error codes (like `TS2339`) to
+ * a `Set` of all the messages we see for that code
+ */
+const getErrorCodeMessages = () => {
+  let errorCodeMessageMap = {};
+
+  prData.forEach((error) => {
+    let errorCode = error.value.tsError.value.errorString;
+    let message = error.value.message.value;
+    errorCodeMessageMap[errorCode] = (errorCodeMessageMap[errorCode] ?? new Set()).add(message);
+  });
+
+  return errorCodeMessageMap;
+};
+
+const errorCodeMessages = getErrorCodeMessages();
+
+/**
+ * Build a map which just counts the entries in an array
+ */
+const countArrayEntries = (xs) => {
```

Comment:
Minor, can we give this parameter a more descriptive name than `xs`? I'm not sure what it's supposed to represent at first glance

Reply:
old habit from Haskell! the idea is if you have a type `x` then the plural of that is `xs`, and `x[]` is something like the plural of `x`...

I'll change it to `arr` or something like that